fdbclient/ReadYourWrites.h

```diff
@@ -54,6 +54,26 @@ struct TransactionDebugInfo : public ReferenceCounted<TransactionDebugInfo> {
 	TransactionDebugInfo() : transactionName(""), lastRetryLogTime() { }
 };
 
+// Private keyspace framework
+class PrivateKeyRangeImpl {
+public:
+	virtual Future<Standalone<RangeResultRef>> getRange(const KeyRange& keys, ReadYourWritesTransaction* ryw) = 0;
+};
+
+class PrivateKeySpace {
+public:
+	static Future<Optional<Value>> get(const Key& key, bool snapshot = false);
+
+	static Future< Standalone<RangeResultRef> > getRange( KeySelector begin, KeySelector end, GetRangeLimits limits, bool snapshot = false, bool reverse = false );
+
+	static void registerKeyRange(const KeyRange& kr, PrivateKeyRangeImpl* impl) {
+		impls.insert(kr, impl);
+	}
+
+private:
+	static KeyRangeMap<PrivateKeyRangeImpl*> impls;
```

Comment:
I don't think anything in here should be static. This might be a problem in case you connect to multiple clusters. Instead this class should be instantiated in `DatabaseContext`

Reply:
Thanks. Done.